OpenRA.Mods.Common/Traits/Transforms.cs

```diff
@@ -127,11 +132,23 @@ public void DeployTransform(bool queued)
 				Faction = faction
 			};
 
-			var makeAnimation = self.TraitOrDefault<WithMakeAnimation>();
+			var checkAndTransform = new CallFunc(() =>
+			{
+				// If the deploy fails, try to salvage the situation by reversing the unmake animation.
+				if (CanDeploy(false))
+					self.QueueActivity(transform);
+				else
+				{
+					// Pass null for the second parameter so we don't issue pointless BuildingComplete notifications.
+					if (makeAnimation != null)
+						makeAnimation.Forward(self, null);
+				}
+			});
+
 			if (makeAnimation != null)
```

Comment:
That only happens for unmake animations, and only if you unmake from a building to an actor that is bigger than that building. Make animations aren't played by  `Transforms`, just unmake animations.